### ISSUE TITLE
Testing approach?

### ISSUE THREAD

**Issue by benjaminkay93** [NaSTAuk/graphics#3]
This could get big, and software engineers in industry will have robust testing to make sure code is production ready before publishing. However, this may give us a barrier to entry to inexperienced developers who want to get involved but may feel overwhelmed by testing.

Should we implement this with testing?

What mitigations can we do programmatically to ensure these are nice and easy for people to pick up if we do decide to put in testing?

**Comment by moschopsuk** [comment]
Testing in the core will be a must.

I think we may have to reach some sort of compromise with components that some may not have tests and as such its a risk you may have to take.

**Comment by benjaminkay93** [comment]
true, I was thinking maybe some sort of snapshot tests to visual check for regressions, and not much more? plus I can see a world where it is easy to set up for contributors and have some level of confidence on the code they have wrote, without much complexity :) 

I do agree the core systems and mechanisms would be very poor without a comprehensive testing sweat.

**Comment by niklasR** [comment]
I like the concept of snapshots.
I think like Luke @moschopsuk said, some core functionality should be tested always, but if there is different templated with special edge cases (e.g. animations, or a crazy amount of variation/literal edge cases with certain rules in sports or so), the template author should make sensible decisions at their own risk and communicate those?

**Comment by benjaminkay93** [comment]
Animations are the one I think we'll find hardest to test. I honestly have no idea where to start on anything like that. There is only so much we can do considering how edge casey some things can be.

I remember making some water polo graphics. The sheer amount of variations made it a headache.
